software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -151,8 +151,18 @@ public void attachLifecycleEffectors(Entity entity) {
                 .build();
     }
 
+    /** @see {@link #newStartEffector()} */
+    public Effector<Void> newSuspendEffector() {
+        return Effectors.effector(Void.class, "suspend")
+                .description("Suspend the process/service represented by an entity")
+                .parameter(StopSoftwareParameters.STOP_PROCESS_MODE)
+                .parameter(StopSoftwareParameters.STOP_MACHINE_MODE)
+                .impl(newSuspendEffectorTask())
+                .build();
+    }
```

Comment:
What are the semantics of `stopMachineMode` here? Would `IF_NOT_STOPPED` really mean `IF_NOT_SUSPENDED`? How does this interact with `stop`? Is a suspended machine stoppable via `stop(IF_NOT_STOPPED)`?